ios/RnExecutorch/models/StyleTransferModel.mm

```diff
@@ -1,88 +1,46 @@
 #import "StyleTransferModel.h"
 #import "../utils/ImageProcessor.h"
+#import "opencv2/opencv.hpp"
 
-@implementation StyleTransferModel
+@implementation StyleTransferModel {
+  cv::Size originalSize;
+}
 
-- (float *) NSArrayToFloatArray:(NSArray<NSNumber *> *)array outLength:(size_t )outLength {
-  if (!array || array.count == 0) {
-    NSLog(@"Invalid NSArray input.");
-    outLength = 0;
-    return nullptr;
-  }
-  
-  size_t length = array.count;
-  float* floatArray = new float[length];
+- (cv::Size)getModelImageSize{
+  NSArray * inputShape = [module getInputShape: @0];
+  NSNumber *widthNumber = inputShape.lastObject;
+  NSNumber *heightNumber = inputShape[inputShape.count - 2];
   
-  for (size_t i = 0; i < length; ++i) {
-    floatArray[i] = [array[i] floatValue];
-  }
+  int height = [heightNumber intValue];
+  int width = [widthNumber intValue];
   
-  outLength = length;
-  return floatArray;
+  return cv::Size(height, width);
 }
 
-- (NSArray *) floatArrayToNSArray:(float*) floatArray length:(size_t)length {
-  if (floatArray == nullptr || length == 0) {
-    NSLog(@"Invalid input array or length.");
-    return nil;
-  }
-  
-  NSMutableArray *array = [NSMutableArray arrayWithCapacity:length];
+- (NSArray *)preprocess:(cv::Mat &)input {
+  self->originalSize = cv::Size(input.cols, input.rows);
   
-  for (size_t i = 0; i < length; ++i) {
-    NSNumber *number = [NSNumber numberWithFloat:floatArray[i]];
-    [array addObject:number];
-  }
+  cv::Size modelImageSize = [self getModelImageSize];
+  cv::resize(input, input, modelImageSize);
```

Comment:
never mind this, I thought initially this is an array pointer and not a `cv::Mat` object. Anyway since this is not performed 'inplace' due to the above reason I would consider creating another `cv::Mat` object `output` for better readability